thesis/Chapters/Chapter1.tex

```diff
@@ -0,0 +1,102 @@
+We can solve this by automating the process, however it's not as simple as just building the frontend assets and uploading them to the cloud.
+
+The operations and sales teams must be able to create deployments on their own and they must be able to control certain aspects of the final product.
+
+
+\subsection{Asset managment}
+
+Brand images, texts, ...
+
+This is currently kept in a CMS. CMS' are made for this, which automatically gives us a lot of functionality. 
+
+An additional pain point here is that the structure of the data is still manually managed. Stampix has asked to see if there is some sort of IaC solution possible for this.
+
+\subsection{Deployment configuration}
+
+All the following is currently stored inside the CMS but define a configuration of the deployment.
+
+\begin{itemize}
+	\item Domain name (company.stampix.com).
+	\item Campaign ID (corresponds to an ID in the database)
+	\item Demo mode (No real codes needed but nothing gets printed either)
+	\item googleAnalyticsTrackingCode
+	\item facebookPixelId
+\end{itemize}
+
+Ideally, we can keep track of these variables separately from the regular assets. While the distinction is hard to make right now, some of these values are needed outside the statically generated app. For example, we'll need to domain name to set up DNS.
+
+This should be a (semi) structured document which can be easily edited by non-development teams and finally sent to our IaC solution to create deployments.
+
+\subsection{Deployment stages}
+
+There should be a clear distinction between deployments in production, staging or test. 
+A common occurence right now is that the sales team will create demo's to use during their pitches. This has the risk that they change some configuration which breaks production apps. These demo's also get made, pitched and then promptly forgotten, never to be used again.
+
+\begin{itemize}
+	\item Production: deployments that are live, in actual use.
+	\item Staging: deployments before going to production. Final checks happen here
+	\item Test: sales demos, development builds, \dots anything else
+\end{itemize}
+
+\subsection{Modularity}
+
+The web app must support 'plugins'. These plugins can literally be anything. They can include extra logic in the backend, extra pages in the frontend or a combination of both.
+
+TODO: How will this be configured? Could be a multi-select in the deployment configuration
+
+\subsubsection{NSFW check}
+
+This plugin checks every order for photos that has \textbf{N}ot \textbf{S}afe \textbf{F}or \textbf{W}ork content. If an order includes content like this, it gets rejected.
+
+\subsubsection{Thank you module}
+
+After creating an order, the user is presented with a small form that asks if they would like to say thank you. The user can select an emoji to reflect their feelings. 
+This module is particularly useful for creating metrics at the end of a campaign
```

Comment:
Same here, lot's of stuff that's already defined, lmk if I can help out with that